Gems/Atom/Asset/Shader/Code/Source/Editor/CommonFiles/Preprocessor.cpp

```diff
@@ -392,16 +470,24 @@ namespace AZ
             AZ_Warning(builderName, success, "Preprocessor option: Could not acquire a list of scan folders from the database.");
 
             // but while we transfer to the set, we're going to keep only folders where +/ShaderLib exists
-            for (AZ::IO::Path shaderScanFolder : scanFoldersVector)
+            for (AZ::IO::Path inShaderScanFolder : scanFoldersVector)
             {
-                shaderScanFolder /= "ShaderLib";
+                auto shaderScanFolder = inShaderScanFolder / "ShaderLib";
                 if (auto it = AZStd::find_if(includePaths.begin(), includePaths.end(), PathCompare(shaderScanFolder));
                     it == includePaths.end())
                 {
                     // the folders constructed this fashion constitute the base of automatic include search paths
                     if (AZ::IO::SystemFile::Exists(shaderScanFolder.c_str()))
                     {
                         includePaths.emplace_back(AZStd::move(shaderScanFolder.LexicallyNormal().Native()));
+
+                        if (inShaderScanFolder == projectPath)
+                        {
+                            // the scenesrg.srgi and viewsrg.srgi needs to include SceneSrgIncludesAll.azsli / ViewSrgIncludesAll.azsli
+                            // before the first occurence of the partial srg. To keep old projects working with the new version, simply
+                            // patch the include into these files.
+                            PatchSrgiIncludeFiles(shaderScanFolder);
```

Comment:
I am really not a fan of patching an existing file to match new requirement of the engine as :
- It is fragile, can fail easily as user change the file
- Won't work if you use perforce source control as files are readonly when not checked-out
- When we want to update the default project ShaderLib file again, will need to take more case into account, this will grow in complexity and become unmaintainable.

What I propose instead, is :
- We hardcode a Hash of the current default ShaderLib folder (as it was before your original change)
- In this if condition, do a hash of the local folder. 
    - If folder is not there or hash match the hardcoded one, use a default ShaderLib folder located in the engine side (which will be able to evolve as engine version)
    - If hash does not match the hardcoded one, means user modified the folder, so load it

Then also need to remove the ShaderLib folder from all of the default templates. As a general rule of thumb in my opinion, we should remove as much content as possible from project templates to prevent these cases and be cornered out when we want to upgrade something

Reply:
Also as you kinda proposed this solution in the [discord thread](https://discord.com/channels/805939474655346758/1313552538850824243/1313892776521371680), this is my awnser to the cons:
- `Behaviour would be very obtuse for the user: e.g. adding a whitespace to the scenesrg.srgi, even by accident, would lead to errors.` 
-> I don't expect anyone from modifying a file like this by accident (and when critical pipeline build error occurs, can still add an explicit error message unlike now to hint that something might be up in the local ShaderLib file), furthermore as new engine version rolls out and become standard I can remove these files from the existing samples to reduce the risk even more
- "Custom .azsli files (i.e. unrelated to srgi) placed in that folder suddenly aren't found anymore"
-> It would mean that the hash does not match, so we would load the folder content